### ISSUE TITLE
bug: Restarting docker container stuck on replaying s3 API calls when presigned PUT url was used

### ISSUE THREAD

**Issue by hsimpson** [localstack/localstack#4655]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When using `DATA_DIR` for persistence and restarting the docker container it is hanging with:
```shell
Waiting for all LocalStack services to be ready
Waiting for all LocalStack services to be ready
Waiting for all LocalStack services to be ready
...
```
forever.
This happens as soon as a presigned PUT Url is used by a client (Chrome) or uploading an object into a bucket.
When using the the awslocal CLI to put an object:
`awslocal s3api put-object --bucket <MY_BUCKENT_NAME> --region eu-central-1 --key <ANY_KEY> --body <ANY_FILE>`
It does not hang during replay.

How the presigning is made:
```typescript
import {
  PutObjectCommand,
  S3Client,
  S3ClientConfig,
} from '@aws-sdk/client-s3';

import { getSignedUrl } from '@aws-sdk/s3-request-presigner';

async function test() {

  const s3ClientConfig: S3ClientConfig = {
    credentials: {
      accessKeyId: 'test',
      secretAccessKey: 'test',
    },
    region: 'eu-central-1',
    endpoint: 'http://localstack:4566',
    forcePathStyle: true
  };
  
  const s3Client = new S3Client(s3ClientConfig);
  
  const command = new PutObjectCommand({
    Bucket: 'my_bucket_name',
    Key: 'my_key',
  });
  
  let url = await getSignedUrl(s3Client, command, {
    expiresIn: 3600,
  });

  url = url.replace('localstack', 'localhost');
  console.log(url);
}
```

### Expected Behavior

The container should replay all S3 api commands

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

Localstack container config from docker-compose.yml:
```yaml
localstack:
    image: localstack/localstack:latest
    restart: always
    ports:
      - '4563-4599:4563-4599'
      - '8055:8080'
    environment:
      - SERVICES=s3
      - DEBUG=1
      - S3_SKIP_SIGNATURE_VALIDATION=1
      - DATA_DIR=/tmp/localstack/data
      - AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
      - AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
      - AWS_REGION=${AWS_REGION}
      - AWS_BUCKET_NAME=${AWS_BUCKET_NAME}
    networks:
      - localstack

    volumes:
      - ./localstack/tmp:/tmp/localstack
      - ./localstack/init:/docker-entrypoint-initaws.d
      - '/var/run/docker.sock:/var/run/docker.sock'
```

There is an init script in "./localstack/init":
```shell
#!/bin/bash
set -x
# awslocal s3 mb "s3://$AWS_BUCKET_NAME"
awslocal s3api create-bucket  --bucket "$AWS_BUCKET_NAME" --region "$AWS_REGION"
set +x
```

### Environment

```markdown
- OS: macOS Big Sur 11.6 (on Apple M1)
- LocalStack: latest
```

**Comment by wvargas-ods** [comment]
I have the same problem, any idea?

**Comment by hsimpson** [comment]
This themes to be fixed with 0.12.19.1 Docker image version
